File: python/7.py
class Solution:
    def reverse(self, x: int) -> int:
        if x > 0:
            res = int(str(x)[::-1])
        else:
            res = -1 * int(str(abs(x))[::-1])
            
        if res > 2**31 - 1 or res < -2**31:
            return 0
        else:
            return res

File: C#/7.cs
public class Solution {
    public int Reverse(int x) {
        int rev = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (rev > int.MaxValue/10 || (rev == int.MaxValue / 10 && pop > 7)) return 0;
            if (rev < int.MinValue/10 || (rev == int.MinValue / 10 && pop < -8)) return 0;
            rev = rev * 10 + pop;
        }
        return rev;
    }
}